filebeat/harvester/reader/multiline.go

```diff
@@ -137,6 +149,11 @@ func (mlr *Multiline) readFirst() (Message, error) {
 
 func (mlr *Multiline) readNext() (Message, error) {
 	for {
+		select {
+		case <-mlr.done:
+			return Message{}, ErrReaderStopped
+		default:
+		}
```

Comment:
TODO: have reader check the error code from `mlr.reader.Next()`. Let's return error immediately instead of pushing half-finished event before pushing actual error.